### PR TITLE
Add icu75 for NodeJS and ICU4C execution

### PR DESCRIPTION
**Pull request by sven-oly**
Uses ICU75.1 and CLDR 45 data. Now there will be 7 versions of NodeJS in the data.

### CODE REVIEW

**Comment by sven-oly** [comment]
PTAL. I'd love to get this checked in soon.

File: testdriver/datasets.py [hunk 1]
```diff
@@ -208,7 +208,8 @@ class ParallelMode(Enum):
   ParallelByLang = 2
 
 class NodeVersion(Enum):
-  Node20 = "20.1.0"
+  Node22 = "22.1.0"
+  Node20 = "21.6.0"
```
Comment (echeran) [inline testdriver/datasets.py:212]: Why does the `Node20` string begins with "21"? I don't see any usages of `Node22` in this PR, for example, so is that addition necessary, or can we remove `Node22`. And for that matter, where else does `NodeVersion` get used, or can we remove that altogether?

File: testgen/generators/common.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+
+function xsample_tests(all_tests, run_limit) {
```
Comment (echeran) [inline testgen/generators/common.js:2]: Is the spelling of `xsample_tests` intentional?

File: testgen/generators/datetime_gen.js [hunk 3]
```diff
@@ -260,6 +285,19 @@ function generateAll(run_limit) {
           all_options['timeZone'] = timezone;
         }
         if (number_system) {
+          try {
+            const common_number_systems = intl_locale.numberingSystems;
+            const supported_calendars = intl_locale.calendars;
+            // TODO: Check if the number system is supported in this locale.
```
Comment (echeran) [inline testgen/generators/datetime_gen.js:291]: Is it easy enough to implement the code for the TODO task re: the number system?

**Review by echeran — APPROVED** [review]

File: testgen/generators/common.js (hunk 2 above)
Comment (sven-oly) [reply testgen/generators/common.js:2]: Fixing this. Thx.

File: testdriver/datasets.py (hunk 1 above)
Comment (sven-oly) [reply testdriver/datasets.py:212]: Removing NodeVersion. Good catch!

File: testgen/generators/datetime_gen.js (hunk 3 above)
Comment (sven-oly) [reply testgen/generators/datetime_gen.js]: Updated this. PTAL.

File: testgen/generators/plural_rules_gen.js [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+const styles = ['long', 'short', 'narrow'];
+
+const lists = [
+  [],
+  ['cat'],
+  ['dog', 'cat'],
+  ['fish', 'mouse', 'parakeet'],
+  ['3 weeks', '2 days', '3 hours', '25 minutes']
+];
```
Comment (sffc) [inline testgen/generators/plural_rules_gen.js:29]: Double-check this file; it looks like it generates list data, but:

1. It is named `plural_rules_gen.js`
2. It is in a PR that should be a clean ICU 75 update

**Review by sffc — APPROVED** [review]

File: testgen/generators/plural_rules_gen.js (hunk 4 above)
Comment (sven-oly) [reply testgen/generators/plural_rules_gen.js]: Removed this generator. Thx for catching this.
